### PR TITLE
docs(slash): Update notes on backend modes

### PR DESCRIPTION
**Pull request by bucanero**
Fixes DOC-143

### CODE REVIEW

File: content/admin/backend-modes.md [hunk 1]
```diff
@@ -21,6 +26,11 @@ can use DQL (formerly *GraphQL+-*) queries and mutations in this mode, as
...
 queries and mutations must be valid for the applied GraphQL schema.
 
+{{% notice "note" %}}
+In GraphQL mode, Slash will enforce a [strict schema](https://dgraph.io/docs/deploy/dgraph-administration/#restricting-mutation-operations), allowing mutations only on predicates already present in the schema.
+Before executing a mutation on a predicate that doesn’t exist in the schema, you need to perform an `alter` operation with that predicate and its schema type.
+{{% /notice %}}
+
 ### Flexible Mode
 
 Flexible mode is suitable for users who are already familiar with Dgraph, and
```
Comment (anand-chandrashekar) [inline content/admin/backend-modes.md:36]: In the ## Flexible Mode section, Can we please clearly note that Dgraph will run without enforcing the 'strict schema" (which we noted in the GraphQL mode).

File: content/admin/backend-modes.md [hunk 2]
```diff
@@ -8,6 +8,11 @@ weight = 6
 Slash GraphQL supports three different backend modes, which control how the
 underlying Dgraph instance is configured.
```
Comment (anand-chandrashekar) [inline content/admin/backend-modes.md:9]: We can add that:
  
Each mode is pre-configured keeping ease of use in mind. If your use case involves querying a read-only dataset, then you can switch on Read-Only Mode. If your use case is centered around schemas and queries based on GraphQL, then GraphQL mode is more suitable for you. Finally, Flexible mode supports both GraphQL and DQL, and is often preferred by users who are more familiar with DQL.

Comment (bucanero) [reply content/admin/backend-modes.md]: added 👍

**Review by anand-chandrashekar — APPROVED** [review]
Approved

**Pull request merged**